core/src/main/java/hudson/tasks/BatchFile.java

```diff
@@ -102,27 +102,19 @@ public String getDisplayName() {
          */
         @Restricted(DoNotUse.class)
         public FormValidation doCheckUnstableReturn(@QueryParameter String value) {
-            value = Util.fixEmptyAndTrim(value);
-            if (value == null) {
-                return FormValidation.ok();
-            }
-            long unstableReturn;
-            try {
-                unstableReturn = Long.parseLong(value);
-            } catch (NumberFormatException e) {
-                return FormValidation.error(hudson.model.Messages.Hudson_NotANumber());
-            }
-            if (unstableReturn == 0) {
-                return FormValidation.warning(hudson.tasks.Messages.BatchFile_invalid_exit_code_zero());
-            }
-            if (unstableReturn < Integer.MIN_VALUE || unstableReturn > Integer.MAX_VALUE) {
-                return FormValidation.error(hudson.tasks.Messages.BatchFile_invalid_exit_code_range(unstableReturn));
-            }
-            return FormValidation.ok();
+            return helpCheckUnstableReturn(value);
         }
 
         public boolean isApplicable(Class<? extends AbstractProject> jobType) {
             return true;
         }
     }
+
+    static String invalidExitCodeZero() {
```

Comment:
Need to remove this strategy as it won't do what I want, per https://github.com/jenkinsci/jenkins/pull/4034#discussion_r294654924